app/models/user_role.rb

```diff
@@ -11,6 +11,18 @@ class UserRole < ActiveRecord::Base
   FACILITY_STAFF = "Facility Staff".freeze
   FACILITY_SENIOR_STAFF = "Facility Senior Staff".freeze
 
+  module AssociationExtension
+
+    def operator?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_roles) }
+    end
+
+    def manager?(facility)
+      any? { |ur| ur.facility == facility && ur.in?(facility_management_roles) }
+    end
+
```

Comment:
What I'm seeing is there sometimes lots of alls to `manager?` and it's does a lot of `SELECT * FROM user_roles where user_id = ?`s. Most of them is hitting the cache, but it still comes up as a noticeable performance difference to not hit the cache (it is possible it's just the logging, so it wouldn't change anything in production, but it slows down in development, where I notice it).

I think what you're proposing would still generate a lot of SQL calls (hopefully mostly cached). The extensions works because the `user_roles` gets memoized after it's first call.